Cabal/Distribution/Backpack/ConfiguredComponent.hs

```diff
@@ -189,23 +198,23 @@ toConfiguredComponent'
     -> Bool -- deterministic
     -> Flag String      -- configIPID (todo: remove me)
     -> Flag ComponentId -- configCID
-    -> Map PackageName (ComponentId, PackageId) -- external
     -> ConfiguredComponentMap
     -> Component
     -> ConfiguredComponent
 toConfiguredComponent' use_external_internal_deps flags
                 pkg_descr deterministic ipid_flag cid_flag
-                external_lib_map (lib_map, exe_map) component =
+                (lib_map, exe_map) component =
     let cc = toConfiguredComponent
                 pkg_descr this_cid
-                external_lib_map (lib_map, exe_map) component
+                (lib_map, exe_map) component
     in if use_external_internal_deps
         then cc { cc_public = True }
         else cc
   where
     this_cid = computeComponentId deterministic ipid_flag cid_flag (package pkg_descr)
                 (componentName component) (Just (deps, flags))
-    deps = [ cid | (cid, _) <- Map.elems external_lib_map ]
+    deps = [ cid | ((dep_pn, _), (cid, _)) <- Map.toList lib_map
+                 , dep_pn /= packageName pkg_descr ]
```

Comment:
The old code probably wasn't right (if I'm in per-component mode, I should hash over the "internal" deps I feed in), but I don't think it's too bad for this code to be wrong because cabal-install doesn't use it anyway.